spacewalk/certs-tools/mgr_ssl_cert_setup.py

```diff
@@ -300,24 +300,16 @@ def getCertWithText(cert):
 
 
 def getPrivateKey(key):
-    if "-----BEGIN RSA PRIVATE KEY-----" in key:
-        privatekeytype = "rsa"
-    elif "-----BEGIN EC PRIVATE KEY-----" in key:
-        privatekeytype = "ec"
-    else:
-        log_error("Unknown Private Key type")
-        return None
-
     # set an invalid password to prevent asking in case of an encrypted one
     out = subprocess.run(
-        ["openssl", privatekeytype, "-passin", "pass:invalid"],
+        ["openssl", "pkey", "-passin", "pass:invalid", "-text", "-noout"],
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE,
         input=key.encode("utf-8")
     )
 
     if out.returncode:
-        log_error("Invalid {} Key: {}".format(privatekeytype.upper(), out.stderr.decode("utf-8")))
+        log_error("Invalid Key: {}".format(out.stderr.decode("utf-8")))
```

Comment:
When we already on it: should this error message maybe give a hint that the key should be not encrypted?